### PR TITLE
Bug 1167216 Misaligned URL Bar Placeholder

### PR DESCRIPTION
**Pull request by fluffyemily**
Increase the inset of the x coord on the placeholder/text of the text field when editing the URL bar so that the text does not "jump" about

https://bugzilla.mozilla.org/show_bug.cgi?id=1167216

### CODE REVIEW

File: Client/Frontend/Browser/URLBarView.swift [hunk 1]
```diff
@@ -637,11 +637,11 @@ private class CurveView: UIView {
 private class ToolbarTextField: AutocompleteTextField {
     override func textRectForBounds(bounds: CGRect) -> CGRect {
         let rect = super.textRectForBounds(bounds)
-        return rect.rectByInsetting(dx: 5, dy: 5)
+        return rect.rectByInsetting(dx: 10, dy: 5)
```
Comment (rnewman) [inline Client/Frontend/Browser/URLBarView.swift]: Where does the 10 come from? (Where did the 5 come from?!)

Comment (fluffyemily) [inline Client/Frontend/Browser/URLBarView.swift]: Don't know where the original 5 came from - was hard-coded in there - but the increase was to align it to the Label in BrowserLocationView. I tested on several different devices and the offset of 10 for placement of the text aligns exactly with the placement of the constraints on the BrowserLocationView Label.

**Comment by wesj** [comment]
Hmm.. We originally were a bit... half-hazard when it came to consts like this. Now we've been trying to pull them out into widget-specific UX-structs like this one:

https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Browser/URLBarView.swift#L10

I think we should try to do that here and use the same (shared?) constants in both BrowserLocationView and URLBarView. i.e. I see some offset(8) calls in here:

https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Browser/BrowserLocationView.swift#L80

that I'm guessing are related to this problem? Overall, this transition (i.e. flipping between two completely different TextView/Labels is kinda janky anyway. Maybe we should file another bug to improve it more. I always wish there was just one textview for the whole thing...

For numbers used throughout the app, we've actually been putting things in here:
https://github.com/mozilla/firefox-ios/blob/master/Utils/AppConstants.swift

I tried/thought about putting a consistent "padding" value in there at one point (I think it is in there?) but I'm not sure its worth it or been successful.

**Comment by fluffyemily** [comment]
we could, just as easily I guess, have 1 UITextField that we just move from
editable to non-editable when the user taps on it. Would make future
alignment issues cropping up irrelevant.

**Comment by wesj** [comment]
Yeah. If you want to try that here sounds good.

**Comment by fluffyemily** [comment]
@wesj OK. I have finally sorted all the issues and updated the tests and ensured that no functionality is lost. Just a single BrowserLocationView that handles everything to do with browser location

**Pull request merged**